java/code/src/com/redhat/rhn/manager/channel/ChannelManager.java

```diff
@@ -2421,16 +2421,27 @@ public static void disassociateChannelEntries(Channel c) {
     public static void addPackages(Channel chan, List<Long> packageIds, User user) {
 
         if (!UserManager.verifyChannelAdmin(user, chan)) {
-            StringBuilder msg = new StringBuilder("User: ");
-            msg.append(user.getLogin());
-            msg.append(" does not have channel admin access to channel: ");
-            msg.append(chan.getLabel());
-
-            LocalizationService ls = LocalizationService.getInstance();
-            PermissionException pex = new PermissionException(msg.toString());
-            pex.setLocalizedTitle(ls.getMessage("permission.jsp.title.channel"));
-            pex.setLocalizedSummary(ls.getMessage("permission.jsp.summary.channel"));
-            throw pex;
+            List<String> allowedList = Config.get().getList(ConfigDefaults.ALLOW_ADDING_PATCHES_VIA_API);
+            if (chan.isVendorChannel() && allowedList.contains(chan.getLabel())) {
+                log.warn("Channel: " + chan.getLabel() +
```

Comment:
I would suggest to add a bit more descriptive string here, such as "Package IDs ... added to vendor channel..." (just like you did above for the errata editing).